### PR TITLE
Update the readme for Wire 2.0.

### PR DESCRIPTION
**Pull request by swankjesse**
New second motivating protocol buffers.
New, giant code snippet.

This doesn't cover important stuff like the schema API or
extensions. We should probably create a Wire wiki for that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,169 +1,375 @@
-Wire Mobile Protocol Buffers
-============================
+Wire Protocol Buffers
+=====================
 
-*“A man got to have a code!”* - Omar Little
```
Comment (danrice-square) [inline README.md]: You can't get rid of the only Wire reference!

Comment (swankjesse) [inline README.md]: That can be fixed! (Off topic: I really gotta watch the Wire. I'm overdue.)

File: README.md [hunk 2]
```diff
@@ -1,169 +1,375 @@
...
+ * It's focused. Proto schemas describe your data models. That's it.
 
-The `wire-compiler` package contains the `WireCompiler` class, which compiles standard `.proto` files
-into Java source code.
+Here's a sample message definition:
 
-For example, to compile the file `protos-repo/google/protobuf/descriptor.proto`, which may
-(recursively) import other `.proto` files within the `protos-repo/` directory (replace
-"VERSION" with the Wire version you are using):
+    syntax = "proto2";
```
Comment (JakeWharton) [inline README.md]: You can unindent this and wrap with

    ```proto
    ```

for syntax higlighting

File: README.md [hunk 3]
```diff
@@ -1,169 +1,375 @@
...
 
-Snapshots of the development version are available in [Sonatype's `snapshots` repository][snap].
+Protocol Buffers also defines a compact binary encoding of messages that conform to the schema. This
+encoding is fast to encode, fast to decode, small to transmit, and small to store. The binary
+encoding uses numeric tags from the schema, like the `5` for `period` above.
 
-If you use Proguard, then you need to add `keep` rules.  The simplest option is to tell Proguard not to touch the Wire library and your generated protocol buffers (of course these simple rules will miss opportunities to shrink and optimize the code):
+For example, let's encode this dinosaur:
+
+    {
```
Comment (JakeWharton) [inline README.md]:
```json
    ```

(even though it's just an arbitrary object definition)

File: README.md [hunk 4]
```diff
@@ -1,169 +1,375 @@
...
+     67  'g'
+     6f  'o'
+     73  's'
+     61  'a'
+     75  'u'
+     72  'r'
+     75  'u'
+     73  's'
+     28  tag: period(5), field encoding: VARINT(0). 5 << 3 | 0
+     02  JURASSIC(2)
```
Comment (JakeWharton) [inline README.md]: Awesome chart!

**Comment by JakeWharton** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (JakeWharton) [inline README.md]: Both are essential (the quote and viewing the show)

**Pull request merged**